model/config.go

```diff
@@ -161,6 +161,32 @@ func (i *IstioRegistry) RouteRules(namespace string) []*proxyconfig.RouteRule {
 	return out
 }
 
+// RouteRulesBySource selects routing rules from source to destination (sorted by precedence)
+func (i *IstioRegistry) RouteRulesBySource(namespace string, instances []*ServiceInstance) []*proxyconfig.RouteRule {
+	out := make([]*proxyconfig.RouteRule, 0)
+	for _, rule := range i.RouteRules(namespace) {
+		if rule.Match != nil {
+			found := false
+			for _, instance := range instances {
+				if rule.Match.Source != "" && rule.Match.Source != instance.Service.Hostname {
+					continue
```

Comment:
I don't think you need arrays here. It's just the pod filtering out rules before generating the config.